data/refprog/reference-program/src/main/java/com/farao_community/farao/data/refprog/reference_program/ReferenceExchangeData.java

```diff
@@ -6,52 +6,54 @@
  */
 package com.farao_community.farao.data.refprog.reference_program;
 
-import com.powsybl.iidm.network.Country;
-
 /**
  * @author Peter Mitri {@literal <peter.mitri at rte-france.com>}
  */
 public class ReferenceExchangeData {
-    private Country areaOut;
-    private Country areaIn;
+    private ReferenceProgramArea areaOut;
+    private ReferenceProgramArea areaIn;
     private double flow;
 
     /**
      * @param areaOut origin country
      * @param areaIn destination country
      * @param flow flow exchanged from origin country to destination country in MW
      */
-    public ReferenceExchangeData(Country areaOut, Country areaIn, double flow) {
+    public ReferenceExchangeData(ReferenceProgramArea areaOut, ReferenceProgramArea areaIn, double flow) {
         this.areaOut = areaOut;
         this.areaIn = areaIn;
         this.flow = flow;
     }
 
-    public Country getAreaOut() {
+    public ReferenceProgramArea getAreaOut() {
         return areaOut;
     }
 
-    public Country getAreaIn() {
+    public ReferenceProgramArea getAreaIn() {
         return areaIn;
     }
 
     public double getFlow() {
         return flow;
     }
 
-    public void setAreaOut(Country areaOut) {
+    public void setAreaOut(ReferenceProgramArea areaOut) {
         this.areaOut = areaOut;
     }
 
-    public void setAreaIn(Country areaIn) {
+    public void setAreaIn(ReferenceProgramArea areaIn) {
         this.areaIn = areaIn;
     }
 
     public void setFlow(double flow) {
         this.flow = flow;
     }
 
-    boolean isAreaOutToAreaInExchange(Country areaOutId, Country areaInId) {
-        return this.areaIn != null && this.areaOut != null && this.areaIn.equals(areaInId) && this.areaOut.equals(areaOutId);
+    boolean isAreaOutToAreaInExchange(ReferenceProgramArea areaOut, ReferenceProgramArea areaIn) {
+        return this.areaIn != null && this.areaOut != null && this.areaIn.equals(areaIn) && this.areaOut.equals(areaOut);
+    }
+
+    public boolean isVirtualHub() {
+        return areaIn == null || areaIn.isVirtualHub() || areaOut == null || areaOut.isVirtualHub();
     }
```

Comment:
I find this method quite confusing : 
* a ReferenceExchangeData involves two areas, one can be a virtual hub but not the other ... so the name of the method seems a bit strange to me 
* when we build the ReferenceProgram with the `CountryNetPositionComputation` class, we directly computes the net-position of countries (and not the country-to-country exchanges), and therefore build ReferenceExchangeData with {country1="BE", country2=null} ... So I am not sure the `areaOut == null` works as expected in all cases

I would prefer to delete it.

Reply:
I have renamed this method "involvesVirtualHub". It is needed in LoopFlowComputation for the moment, I think we'll be able to delete it once the branch of farao-virtual-hubs is released.